packages/ckeditor5-engine/tests/view/matcher.js

```diff
@@ -246,6 +550,65 @@ describe( 'Matcher', () => {
 			expect( matcher.match( el3 ) ).to.be.null;
 		} );
 
+		it( 'should match element styles using key->value, where both are RegExp objects', () => {
+			const pattern = {
+				styles: [
+					{ key: /border-.*/, value: /.*/ }
+				]
+			};
+			const matcher = new Matcher( pattern );
+			const el1 = new Element( document, 'p', { style: 'border-top: 1px solid blue' } );
+			const el2 = new Element( document, 'p', { style: 'border-top-width: 3px' } );
+			const el3 = new Element( document, 'p', { style: 'color: red' } );
+
+			let result = matcher.match( el1 );
+			expect( result ).to.be.an( 'object' );
+			expect( result ).to.have.property( 'element' ).that.equal( el1 );
+			expect( result ).to.have.property( 'pattern' ).that.equal( pattern );
+			expect( result ).to.have.property( 'match' ).that.has.property( 'styles' ).that.is.an( 'array' );
+			expect( result.match.styles ).to.deep.equal( [
+				'border-color',
+				'border-style',
+				'border-width',
+				'border-top',
+				'border-top-color',
+				'border-top-style',
+				'border-top-width'
+			] );
```

Comment:
I'm wondering if currently we are not matching too much, but I will explain my doubts in a summary comment.

Reply:
I started writing my doubts and find out that I missed the point that we will be passing more strict patterns this match will be much more restricted (like using only `border-top` will only give us `border-top` property, not all of them) so it seems to be fine 👍🏻